src/realm-settings/NewClientPolicyForm.tsx

```diff
@@ -143,29 +145,37 @@ export default function NewClientPolicyForm() {
       "org.keycloak.services.clientpolicy.condition.ClientPolicyConditionProvider"
     ];
 
-  const formValues = form.getValues();
-
   const save = async () => {
-    const createdForm = formValues;
+    const createdForm = form.getValues();
     const createdPolicy = {
       ...createdForm,
       profiles: [],
       conditions: [],
     };
 
-    const policyNameExists = policies.find(
-      (policy) => policy.name === createdPolicy.name
-    );
+    const getAllPolicies = () => {
+      const policyNameExists = !!policies.find(
+        (policy) => policy.name === createdPolicy.name
+      );
 
-    const res = policies.map((policy) =>
-      policy.name === createdPolicy.name ? createdPolicy : policy
-    );
+      const res = policies.map((policy) =>
+        policy.name === createdPolicy.name ? createdPolicy : policy
+      );
 
-    const allPolicies = policyNameExists ? res : policies.concat(createdForm);
+      if (policyNameExists) {
+        return res;
```

Comment:
Seems like `res` is only used in this branch, so perhaps you can return it inline?
```suggestion
        return policies.map((policy) =>
          policy.name === createdPolicy.name ? createdPolicy : policy
        );
```